unit-tests/src/test/scala/java/util/PropertiesTest.scala

```diff
@@ -0,0 +1,448 @@
+    prop.setProperty("a", "A")
+    prop.setProperty("b", "B")
+    prop.setProperty("c", "C")
+    assertEquals(3, prop.stringPropertyNames().size)
+    assertCollSameElementsAsSet("a", "b", "c")(prop.stringPropertyNames())
+
+    val prop2 = new Properties(prop)
+    prop.setProperty("c", "CC")
+    prop.setProperty("d", "D")
+    assertEquals(4, prop2.stringPropertyNames().size)
+    assertCollSameElementsAsSet("a", "b", "c", "d")(prop2.stringPropertyNames())
+  }
+
+  @Ignore @Test def stringPropertyNamesIsNotAffectedByOverriddenStringPropertyNamesInDefaults()
+      : Unit = {
+    val defaults = new java.util.Properties {
+      override def stringPropertyNames(): ju.Set[String] =
+        ju.Collections.emptySet[String]()
+    }
+    defaults.setProperty("foo", "bar")
+
+    val props = new Properties(defaults)
+    props.setProperty("foobar", "babar")
+    assertCollSameElementsAsSet("foo", "foobar")(props.stringPropertyNames())
+  }
+
+  @Test def stringPropertyNamesWithBadContents(): Unit = {
+    assumeTrue("Assumed compliant asInstanceOf", hasCompliantAsInstanceOfs)
+
+    val prop = new Properties()
+    prop.setProperty("a", "A")
+    prop.setProperty("b", "B")
+    prop.setProperty("c", "C")
+
+    prop.put(1.asInstanceOf[AnyRef], "2")
+    assertCollSameElementsAsSet("a", "b", "c")(prop.stringPropertyNames())
+    prop.remove(1.asInstanceOf[AnyRef])
+
+    prop.put("1", 1.asInstanceOf[AnyRef])
+    assertCollSameElementsAsSet("a", "b", "c")(prop.stringPropertyNames())
+    prop.remove("1")
+
+    val prop2 = new Properties(prop)
+    prop.setProperty("c", "CC")
+    prop.setProperty("d", "D")
+
+    prop2.put(1.asInstanceOf[AnyRef], "2")
+    assertCollSameElementsAsSet("a", "b", "c", "d")(prop2.stringPropertyNames())
+    prop2.remove(1.asInstanceOf[AnyRef])
+
+    prop2.put("1", 1.asInstanceOf[AnyRef])
+    assertCollSameElementsAsSet("a", "b", "c", "d")(prop2.stringPropertyNames())
+  }
+
+  // ported from Harmony
+  @Test def put_on_null_key_or_null_value(): Unit = {
+    val properties = new Properties
+    assertThrows(classOf[NullPointerException], properties.put(null, "any"))
+    assertThrows(classOf[NullPointerException], properties.put("any", null))
+  }
+
+  @Test def non_string_values(): Unit = {
+    val properties = new Properties
+
+    properties.put("age", Int.box(18))
+    assertNull(properties.getProperty("age"))
+    assertThrows(classOf[ClassCastException],
+                 properties.list(new PrintWriter(new ByteArrayOutputStream)))
+  }
+
+  @Test def list(): Unit = {
+    val properties = new Properties
+
+    def assertResult(result: String): Unit = {
+      val buffer4stream = new ByteArrayOutputStream
+      val stream        = new PrintStream(buffer4stream)
+      properties.list(stream)
+      assertEquals(buffer4stream.toString.trim, result.trim)
+      stream.flush()
+    }
+
+    val result0 = "-- listing properties --\n"
+    assertResult(result0)
```

Comment:
Any reason you're not directly passing the literal strings as arguments to `assertResult`?

Reply:
I don't see any reason so I changed it.